### PR TITLE
Update Adikteev adapter (PR updated)

### PR DESCRIPTION
**Pull request by piotr-yuxuan**
## Type of change

- [x] Adapter feature update
- [x] This change affects user-facing APIs or examples documented on http://prebid.org.

## Description of change

Test parameters for validating bids

``` javascript
{
    bidder: 'adikteev',
    params: {
       adSpaceId: 5084
    }
}
```

- Integration tested with Adikteev's [Hello World](https://github.com/Adikteev/Prebid.js-old/blob/0cbc11013bf69e744cd81ba9085efcf8e9d3e31d/integrationExamples/gpt/hello_world_adikteev.html) sample page.
- Maintainer: adnetwork@adikteev.com
- [PR on the docs repo](https://github.com/prebid/prebid.github.io/pull/1066)

### CODE REVIEW

File: modules/adikteevBidAdapter.js [hunk 1]
```diff
@@ -1,91 +1,223 @@
...
-      currency: config.getConfig('currency'),
-      userAgent: navigator.userAgent,
-      screen: {
-        width: window.screen.width,
-        height: window.screen.height
+    const payload = Object.assign({},
+      {
+        akPbjsVersion: AK_PBJS_VERSION,
+        bidRequests: validBidRequests.map(conformBidRequest),
+        cookies: conformCookies(document.cookie),
```
Comment (mike-chowla) [inline modules/adikteevBidAdapter.js]: Why are you sending the document's cookies?

Comment (piotr-yuxuan) [reply modules/adikteevBidAdapter.js]: I've removed this part.

File: modules/adikteevBidAdapter.js [hunk 2]
```diff
@@ -1,91 +1,211 @@
...
+import {config} from '../src/config';
 
 export const BIDDER_CODE = 'adikteev';
-export const ENDPOINT_URL = 'https://serve-adserver.adikteev.com/api/prebid/bid';
-export const ENDPOINT_URL_STAGING = 'https://serve-adserver-staging.adikteev.com/api/prebid/bid';
-export const USER_SYNC_IFRAME_URL = 'https://serve-adserver.adikteev.com/api/prebid/sync-iframe';
-export const USER_SYNC_IFRAME_URL_STAGING = 'https://serve-adserver-staging.adikteev.com/api/prebid/sync-iframe';
-export const USER_SYNC_IMAGE_URL = 'https://serve-adserver.adikteev.com/api/prebid/sync-image';
-export const USER_SYNC_IMAGE_URL_STAGING = 'https://serve-adserver-staging.adikteev.com/api/prebid/sync-image';
+export const AK_PBJS_VERSION = '1.35.0';
```
[the remark this answers is not in the document]
Comment (piotr-yuxuan) [reply modules/adikteevBidAdapter.js]: `AK_PBJS_VERSION` flags the version of our adapter. We want it to keep the same value across different `'$prebid.version$'` so our endpoint knows which version of our adapter is calling it. So we don't want Prebid.js version here.

**Comment by piotr-yuxuan** [comment]
@mike-chowla it should be OK and respond with code 200 now. Thanks for your review 🙂

**Comment by piotr-yuxuan** [comment]
My bad @mike-chowla, sorry for the delay. I thought it was only needed to send response code 200 with an (empty) array of bid responses. Now it returns a bid.

**Comment by piotr-yuxuan** [comment]
Hi there,

`master` has seen quite a lot of changes since this branch has been created two months ago, so tests code got broken. I've had to rebase this branch on upstream `master` and repair the tests. In addition to that fix, I've added some additional changes we need.

@mike-chowla In order to ease our integration, would you like to review this PR again? It's been some time we haven't heard from you. I'll be more than happy to do whatever is needed to get this PR merged :-)

Cheers.

File: modules/adikteevBidAdapter.js [hunk 3]
```diff
@@ -4,88 +4,208 @@ import * as utils from '../src/utils';
...
-    const validBidRequests = JSON.parse(bidRequests.data).validBidRequests;
-    serverResponse.body.forEach((value, index) => {
-      const overrides = {
-        requestId: validBidRequests[index].bidId,
-      };
-      returnedBids.push(Object.assign({}, value, overrides));
-    });
-    return returnedBids;
-  },
+  interpretResponse: (serverResponse) => serverResponse.body,
```
Comment (mike-chowla) [inline modules/adikteevBidAdapter.js]: The response that's coming back with the test parameters isn't valid json and is causing the bid to get dropped.

[adikteev_response.txt](https://github.com/prebid/Prebid.js/files/2880536/adikteev_response.txt)

**Comment by mike-chowla** [comment]
Apologies for the slow response.  I've been traveling the last couple of weeks and have gotten behind on my PR reviews

**Comment by piotr-yuxuan** [comment]
Hey! Thank you very much for your reply.

Here is how I test it. I launch `gulp serve` from commit dcca2b6c99f692bc897156a1f4852b2cf7f3490f, then I browse to `http://localhost:9999/integrationExamples/gpt/hello_world_adikteev.html` and I can see an ad displayed in the page.

screenshot 2019-02-19 at 17 55 07

Chrome network dev tools show there is a call to `https://prebid.adikteev.com/api/prebid/bid`, and here is the raw response I've just got:

``` js
[{"netRevenue":true,"width":300,"currency":"EUR","requestId":"2638a6b3ee42ba","creativeId":41601,"ttl":360,"ad":"<div><img height=\"1\" src=\"https://event-adserver.adikteev.com/api/ad_event.json?ad_space_id=5084&amp;sdk_version=8452&amp;what=on_ad_ready_by_sdk&amp;campaign_id=41601&amp;placement_id=46528&amp;device=351d362a-b9ce-4759-af4e-420870263dae&amp;ip=84.14.114.166\" style=\"display:none;visibility:hidden;\" width=\"1\" /><iframe src=\"https://serve-adserver.adikteev.com/api/prebid/serve?ad_space_id=5084&amp;bid_id=2638a6b3ee42ba&amp;sdk_version=9001&amp;placement_id=46528&amp;creative_id=41601\" style=\"background::transparent;border:0;height:100%;margin:0;overflow::hidden;padding:0;width:100%;\"></iframe></div>","cpm":3.325,"height":250}]
```
Once formatted it becomes:
``` js
[{
  "netRevenue": true,
  "width": 300,
  "currency": "EUR",
  "requestId": "2303a14333cf4f",
  "creativeId": 41601,
  "ttl": 360,
  "ad": "<div><img height=\"1\" src=\"https://event-adserver.adikteev.com/api/ad_event.json?ad_space_id=5084&amp;sdk_version=8452&amp;what=on_ad_ready_by_sdk&amp;campaign_id=41601&amp;placement_id=46528&amp;device=e61b94de-4714-4f57-ab3e-6b83f3669f9c&amp;ip=84.14.114.166\" style=\"display:none;visibility:hidden;\" width=\"1\" /><iframe src=\"https://serve-adserver.adikteev.com/api/prebid/serve?ad_space_id=5084&amp;bid_id=2303a14333cf4f&amp;sdk_version=9001&amp;placement_id=46528&amp;creative_id=41601\" style=\"background::transparent;border:0;height:100%;margin:0;overflow::hidden;padding:0;width:100%;\"></iframe></div>",
  "cpm": 3.325,
  "height": 250
}]
```

Am I missing something?

**Comment by piotr-yuxuan** [comment]
It seems like `adikteev_response.txt` content doesn't come from us.

**Comment by piotr-yuxuan** [comment]
Anything I can do to help review this PR? I know it's not enough for this PR to WorkOnMyMachine® 😉 but I'm unsure what I can do more.

**Comment by mike-chowla** [comment]
I tried this again but still getting no bids back.

Request URL: https://prebid.adikteev.com/api/prebid/bid

Request Body:
[adikteev_request.txt](https://github.com/prebid/Prebid.js/files/2892353/adikteev_request.txt)

**Comment by piotr-yuxuan** [comment]
OK, I believe I got it. The ad set up for Prebid was restricted in some countries. Please accept my apologies for that! At least it demonstrates we run our actual ad engine and give an actual response, not a canned one 🙂

**Comment by mike-chowla** [comment]
There's a CORS issue when retreiving the bid:

screen shot 2019-02-26 at 9 03 54 am

**Comment by piotr-yuxuan** [comment]
Well, this error 500 was because of the update of some campaign. Should be alright now.

**Comment by piotr-yuxuan** [comment]
Please let me know if you experience any hindrance whilst reviewing this PR, I'll be most glad to help you.

**Comment by piotr-yuxuan** [comment]
@mike-chowla 😉

**Comment by piotr-yuxuan** [comment]
up?

File: test/spec/modules/adikteevBidAdapter_spec.js [hunk 4]
```diff
@@ -20,21 +74,56 @@ describe('adikteevBidAdapter', () => {
...
     });
-    it('exists and is a function', () => {
-      expect(setstagingEnvironmentSwitch).to.exist.and.to.be.a('function');
-    });
-    it('exists and is correctly set', () => {
-      expect(stagingEnvironmentSwitch).to.exist.and.to.equal(false);
-    });
   });
 
+  describe('conformBidRequest', () => {
```
Comment (jaiminpanchal27) [inline test/spec/modules/adikteevBidAdapter_spec.js]: Can you update your spec file to not use arrow functions. More info here https://github.com/prebid/Prebid.js/issues/2987

**Comment by jeremyvdw** [comment]
Hi @mike-chowla, 
Thanks for your review.
Adikteev is now rebranded as eMoteev: https://twitter.com/Adikteev/status/1105073414103072769, https://www.linkedin.com/feed/update/urn:li:activity:6511615144966135809

We need to address the change in this PR for it to be relevant.
The following commit will reflect this rebranding, no more changes, as we would like to be merged and deployed ASAP.
Thank you.

PS: @jaiminpanchal27 if the change you requested suits you, would you like to approve it? Thx

**Comment by piotr-yuxuan** [comment]
For the sake of completion:

#### POST request to https://prebid.emoteev.io/api/prebid/bid

``` curl
curl 'https://prebid.emoteev.io/api/prebid/bid' -H 'Referer: http://localhost:9999/integrationExamples/gpt/hello_world_emoteev.html' -H 'Origin: http://localhost:9999' -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3736.0 Safari/537.36' -H 'Content-Type: text/plain' --data-binary '{"akPbjsVersion":"1.35.0","bidRequests":[{"params":{"adSpaceId":5084},"crumbs":{"pubcid":"6e040aab-46bd-45bc-b898-e1b72b62d75c"},"sizes":[[300,250],[250,300],[300,600]],"bidId":"2988189446fab3","bidderRequestId":"1d9d620ba0fc0c"}],"debug":false,"language":"en-GB","refererInfo":{"referer":"http://localhost:9999/integrationExamples/gpt/hello_world_emoteev.html","reachedTop":true,"numIframes":0,"stack":["http://localhost:9999/integrationExamples/gpt/hello_world_emoteev.html"]},"deviceInfo":{"browserWidth":1440,"browserHeight":150,"deviceWidth":1440,"deviceHeight":900,"documentWidth":1440,"documentHeight":150,"webGL":true},"userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/75.0.3736.0 Safari/537.36"}' --compressed
```

#### Beautified response

``` json
[
  {
    "width": 300,
    "ad": "<div><img height=\"1px\" src=\"https://event-adserver.adikteev.com/api/ad_event.json?ad_space_id=5084&amp;sdk_version=8481&amp;what=on_ad_ready_by_sdk&amp;campaign_id=41601&amp;placement_id=58710&amp;device=1733e013-61aa-4063-b153-10ebdb360654&amp;ip=84.14.114.166\" style=\"display:none;visibility:hidden;\" width=\"1px\" /><iframe src=\"https://serve-adserver.adikteev.com/api/prebid/serve?ad_space_id=5084&amp;uid=136a7411-11a9-441d-a0d8-5c3e120ef0fd&amp;sdk_version=8481&amp;placement_id=58710&amp;creative_id=41601\" style=\"background:transparent;border:0;height:100%;margin:0;overflow:hidden;padding:0;width:100%;\"></iframe></div>",
    "height": 250,
    "currency": "EUR",
    "netRevenue": true,
    "requestId": "2988189446fab3",
    "ttl": 360,
    "cpm": 3.325,
    "creativeId": 41601
  }
]
```

File: test/spec/modules/adikteevBidAdapter_spec.js (hunk 4 above)
Comment (piotr-yuxuan) [reply test/spec/modules/adikteevBidAdapter_spec.js]: Hey @jaiminpanchal27, done in https://github.com/prebid/Prebid.js/pull/3391/commits/542b03f04e51fdc57506a648a3ac5c7798e7ecd8 🙂 If these changes look good to you, would you mind approve these PR?

Comment (jeremyvdw) [reply test/spec/modules/adikteevBidAdapter_spec.js]: @jaiminpanchal27 I believe this is patched, do you mind approving?

**Pull request merged**